doc/source/cluster/cloud.rst

```diff
@@ -272,6 +272,116 @@ There are two ways of running private clusters:
+
+This section assumes that you have a list of machines and that the nodes in the cluster can communicate with each other. It also assumes that Ray is installed
+on each machine. To install Ray, follow the `installation instructions`_.
+
+.. _`installation instructions`: http://docs.ray.io/en/master/installation.html
+
+Starting Ray on each machine
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+On the head node (just choose some node to be the head node), run the following.
+If the ``--port`` argument is omitted, Ray will choose port 6379, falling back to a
+random port.
+
+.. code-block:: bash
+
+  $ ray start --head --port=6379
+  ...
+  Next steps
+    To connect to this Ray runtime from another node, run
+      ray start --address='<ip address>:6379' --redis-password='<password>'
+
+  If connection fails, check your firewall settings and network configuration.
+
+The command will print out the address of the Redis server that was started
+(the local node IP address plus the port number you specified).
+
+**Then on each of the other nodes**, run the following. Make sure to replace
+``<address>`` with the value printed by the command on the head node (it
+should look something like ``123.45.67.89:6379``).
+
+Note that if your compute nodes are on their own subnetwork with Network
+Address Translation, to connect from a regular machine outside that subnetwork,
+the command printed by the head node will not work. You need to find the
+address that will reach the head node from the second machine. If the head node
+has a domain address like compute04.berkeley.edu, you can simply use that in
+place of an IP address and rely on the DNS.
+
+.. code-block:: bash
+
+  $ ray start --address=<address> --redis-password='<password>'
+  --------------------
+  Ray runtime started.
+  --------------------
+
+  To terminate the Ray runtime, run
+    ray stop
+
+If you wish to specify that a machine has 10 CPUs and 1 GPU, you can do this
+with the flags ``--num-cpus=10`` and ``--num-gpus=1``. See the :ref:`Configuration <configuring-ray>` page for more information.
+
+If you see ``Unable to connect to Redis. If the Redis instance is on a
+different machine, check that your firewall is configured properly.``,
+this means the ``--port`` is inaccessible at the given IP address (because, for
+example, the head node is not actually running Ray, or you have the wrong IP
+address).
+
+If you see ``Ray runtime started.``, then the node successfully connected to
+the IP address at the ``--port``. You should now be able to connect to the
+cluster with ``ray.init(address='auto')``.
+
+If ``ray.init(address='auto')`` keeps repeating
+``redis_context.cc:303: Failed to connect to Redis, retrying.``, then the node
+is failing to connect to some other port(s) besides the main port.
+
+.. code-block:: bash
+
+  If connection fails, check your firewall settings and network configuration.
+
+If the connection fails, to check whether each port can be reached from a node,
```

Comment:
" check whether each port on the head can be reached from the worker node in question, you can use ..."

just so it's clearer what is being detected